### PR TITLE
Discovery coverage for rule priority/limit/provisioning

### PR DESCRIPTION
**Pull request by adarshdubey-star**
Coverage for :
```
test_positive_reboot_all_pxe_hosts
test_positive_end_to_end_crud
test_positive_provision_with_rule_priority
test_positive_multi_provision_with_rule_limit
```
The `reboot_all `scenario is expected to be breaking due to a known issue which will be worked on.

### CODE REVIEW

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_discoveryrule -k test_positive_update_and_provision_with_rule_priority
provisioning: true

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_discoveryrule -k test_positive_update_and_provision_with_rule_priority
provisioning: true

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_discoveryrule.py -k test_positive_update_and_provision_with_rule_priority
provisioning: true

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_discoveryrule -k test_positive_update_and_provision_with_rule_priority
provisioning: true

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_discoveryrule.py -k test_positive_update_and_provision_with_rule_priority
provisioning: true

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_discoveryrule.py -k test_positive_update_and_provision_with_rule_priority
provisioning: true

File: tests/foreman/api/test_discoveredhost.py [hunk 1]
```diff
@@ -402,25 +401,49 @@ def test_positive_reboot_pxe_host(
         result = sat.api.DiscoveredHost(id=discovered_host.id).reboot()
         assert 'Unable to perform reboot' not in result
 
-    @pytest.mark.stubbed
+    # @pytest.mark.on_premises_provisioning
```
Comment (shubhamsg199) [inline tests/foreman/api/test_discoveredhost.py:404]: Don't we need this marker?

File: tests/foreman/api/test_discoveryrule.py [hunk 2]
```diff
@@ -130,44 +97,67 @@ def test_positive_provision_with_rule_priority():
...
+    result = module_target_sat.api.DiscoveredHost(id=discovered_host['id']).auto_provision()
+    assert f'provisioned with rule {prio_rule.name}' in result['message']
 
-    :CaseImportance: High
-    """
+    # Delete discovery rule
+    for _ in rule, prio_rule:
+        _.delete()
+        with pytest.raises(HTTPError):
+            _.read()
```
Comment (shubhamsg199) [inline tests/foreman/api/test_discoveryrule.py:129]: Non blocking: any variable would be good instead of `_`

Comment (adarshdubey-star) [reply tests/foreman/api/test_discoveryrule.py:129]: Any way its just for teardown :)

File: pytest_fixtures/component/provision_pxe.py [hunk 3]
```diff
@@ -237,6 +237,31 @@ def provisioning_host(module_ssh_key_file, pxe_loader):
...
+    """Fixture to check out two blank VMs"""
+    vlan_id = settings.provisioning.vlan_id
+    cd_iso = (
+        ""  # TODO: Make this an optional fixture parameter (update vm_firmware when adding this)
+    )
+
+    with Broker(
+        workflow="deploy-configure-pxe-provisioning-host-rhv",
+        host_class=ContentHost,
+        _count=2,
```
[the remark this answers is not in the document]
Comment (adarshdubey-star) [reply pytest_fixtures/component/provision_pxe.py]: Good idea!

File: tests/foreman/api/test_discoveredhost.py (hunk 1 above)
Comment (adarshdubey-star) [reply tests/foreman/api/test_discoveredhost.py]: Updated

File: tests/foreman/api/test_discoveredhost.py [hunk 4]
```diff
@@ -402,25 +401,51 @@ def test_positive_reboot_pxe_host(
         result = sat.api.DiscoveredHost(id=discovered_host.id).reboot()
         assert 'Unable to perform reboot' not in result
 
-    @pytest.mark.stubbed
+    @pytest.mark.on_premises_provisioning
+    @pytest.mark.parametrize('module_provisioning_sat', ['discovery'], indirect=True)
+    @pytest.mark.parametrize('pxe_loader', ['bios'], indirect=True)
+    @pytest.mark.rhel_ver_match('9')
+    @pytest.mark.parametrize('count', [2])
```
Comment (shubhamsg199) [inline tests/foreman/api/test_discoveredhost.py:408]:
```suggestion
    @pytest.mark.parametrize('provision_multiple_hosts', [2])
```

Comment (adarshdubey-star) [reply tests/foreman/api/test_discoveredhost.py]: yes, I missed that!

**Review by pondrejk — APPROVED** [review]

**Review by shubhamsg199 — APPROVED** [review]
Ack

**Pull request merged**